packages/bierzo-wallet/src/store/reducers.ts

```diff
@@ -15,14 +21,24 @@ export interface RootReducer {
   usernames: UsernamesState;
 }
 
-const createRootReducer = (): Reducer<RootReducer> =>
-  combineReducers({
-    extension: extensionReducer,
-    notifications: notificationReducer,
-    tokens: tokensReducer,
-    balances: balancesReducer,
-    usernames: usernamesReducer,
-  });
+const allReducers = combineReducers({
+  extension: extensionReducer,
+  notifications: notificationReducer,
+  tokens: tokensReducer,
+  balances: balancesReducer,
+  usernames: usernamesReducer,
+});
+
+const createRootReducer = (): Reducer<RootReducer> => (
+  state: RootReducer | undefined,
+  action: ActionType<typeof actions>,
+) => {
+  if (action.type === "@@app/RESET") {
+    return allReducers(undefined, action);
```

Comment:
What does it do? Force to all reducers to load their initial state?

Reply:
Yes

Reply:
@webmaster128 @albertandrejev if the code works, is fine, I was going to add, this is the first time I see implement the logout in this way (maybe for some reason in the latest redux version is recommended), but I think is much more natural and clean to listen in all reducers for the reset action and returns directly the partial state you want there. 

Potential problems I see with this approach:
1. If you add more reducers which you do not want to be modified when the user disconnects you will have problems.
2. If you have strange errors it is not trivial to arrive here and remember the reset stuff.

Anyway, if you guys think this is a safe bet, feel free to leave as it is.

Reply:
But with this approach you do not need to implement reset action in every reducer (in most cases this is simply copy+paste solution and will violate DRY principle). Also you can forget to implement this action and this can cause security leaks. 
If you want you can implement specific behaviour per reducer for reset action, but you do not need to care about it in case if you simply want to return default state from the specific reducer.
You can enhance my solution if you want to leave specific part of the state simply by copying this part into returning state.

For example:
```ts
 if (action.type === "@@app/RESET") {
    const { tokens } = state!;
    const newState = allReducers(undefined, action);
    newState.tokens = { ...tokens };
    return newState;
}
```

You can use this approach and make it more flexible.

Reply:
From Dan Abramov's answer https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store/35641992#35641992